parking_permits_app/models.py

```diff
@@ -0,0 +1,386 @@
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Name"), max_length=32, blank=False, null=False)
+
+    class Meta:
+        db_table = "vehicle_type"
+        verbose_name = _("Vehicle type")
+        verbose_name_plural = _("Vehicle types")
+
+    def __str__(self):
+        return "%s - %s" % (self.id, self.name)
+
+
+class LowEmissionCriteria(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    vehicle_type = models.ForeignKey(
+        VehicleType,
+        verbose_name=_("Vehicle type"),
+        on_delete=models.PROTECT,
+        blank=False,
+        null=False,
+    )
+    emission_limit_nedc = models.IntegerField(
+        _("Emission limit (NEDC)"), blank=True, null=True
+    )
+    emission_limit_wltp = models.IntegerField(
+        _("Emission limit (WLTP)"), blank=True, null=True
+    )
+
+    class Meta:
+        db_table = "low_emission_criteria"
+        verbose_name = _("Low-emission criteria")
+        verbose_name_plural = _("Low-emission criterias")
+
+    def __str__(self):
+        return "%s - %s, NEDC: %s, WLTP: %s" % (
+            self.id,
+            self.vehicle_type,
+            self.emission_limit_nedc,
+            self.emission_limit_wltp,
+        )
+
+
+class Address(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    street_name = models.CharField(
+        _("Street name"), max_length=128, blank=False, null=False
+    )
+    street_number = models.CharField(
+        _("Street number"), max_length=128, blank=False, null=False
+    )
+    city = models.CharField(_("City"), max_length=128, blank=False, null=False)
+    location = models.PointField(
+        _("Location (2D)"), srid=settings.SRID, blank=False, null=False
+    )
+
+    class Meta:
+        db_table = "address"
+        verbose_name = _("Address")
+        verbose_name_plural = _("Addresses")
+
+    def __str__(self):
+        return "%s - %s %s, %s" % (
+            self.id,
+            self.street_name,
+            self.street_number,
+            self.city,
+        )
+
+
+class Company(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Company name"), max_length=128, blank=False, null=False)
+    address = models.ForeignKey(
+        Address, verbose_name=_("Address"), on_delete=models.PROTECT
+    )
+    company_owner = models.ForeignKey(
+        Customer, verbose_name=_("Company owner"), on_delete=models.PROTECT
+    )
+
+    class Meta:
+        db_table = "company"
+        verbose_name = _("Company")
+        verbose_name_plural = _("Companies")
+
+    def __str__(self):
+        return "%s - %s" % (self.id, self.name)
+
+
+class Vehicle(models.Model):
+    CATEGORIES = (
```

Comment:
May I please suggest to move `CATEGORIES` to `constants.py` so that:

1. It'd be more explicit that these are the list of cars that are eligible to buy parking permits using our service.
2. Other features, like upcoming pricing logic, can import and use these categories easily as well.

Reply:
This is a good point and will be updated now.